lib/services/local/access_list.go

```diff
@@ -1029,3 +1045,19 @@ func (a *AccessListService) VerifyAccessListCreateLimit(ctx context.Context, tar
 	const limitReachedMessage = "cluster has reached its limit for creating access lists, please contact the cluster administrator"
 	return trace.AccessDenied(limitReachedMessage)
 }
+
+// keepAWSIdentityCenterLabels preserves member labels if
+// it originated from AWS Identity Center plugin.
+// The Web UI does not currently preserve metadata labels so this function should be called
+// in every update/upsert member calls.
+// Remove this function once https://github.com/gravitational/teleport.e/issues/5415 is addressed.
```

Comment:
Hmm, wouldn't it make more sense to update the [memberToAccessListMember](https://github.com/gravitational/teleport.e/blob/master/lib/web/accesslist.go#L250C1-L269C2) function that you linked in that ticket to keep the labels when creating a new member then?

Reply:
I did look at it before creating the ticket and its not going to be a trivial change. I don't even know if its a feature or a bug cause the way it is currently implemented, its a single API where we upsert entire access list, even if its just a single member addition/removal https://github.com/gravitational/teleport.e/blob/master/lib/web/plugin.go#L285. 

Preserving older state is not something exclusively done for the identity center. See prior work on https://github.com/gravitational/teleport/blob/master/lib/services/local/access_list.go#L729C1-L735C58. Looks like we settled on with the current implementation? Anyway, I think updating that API to handle label and Id is best tackled separately.

Reply:
I understand we replace the list, but from the ticket description the impression I got was that we just do not add `member.Metadata.Labels` in that function. Is that not the case?

I don't have objection to keep this implementation for now but trying to understand why passing the labels to the newly created member in that function does not work.

Reply:
Sorry for not being clear. Another attempt below:
> that we just do not add member.Metadata.Labels in that function. Is that not the case?

Yup correct. But that is also due to the fact that the [memberToAccessListMember](https://github.com/gravitational/teleport.e/blob/master/lib/web/accesslist.go#L250C1-L269C2) function does not have access to the metadata label because we do not send them to the UI in `GET` Access List https://github.com/gravitational/teleport.e/blob/master/lib/web/ui/accesslist.go#L10C1-L25C2. And as a result the UI does not handle metadata values, nor it send them back in the `upsert` request. https://github.com/gravitational/teleport.e/blob/master/lib/web/ui/accesslist.go#L38C1-L42C2. Hence looses any labels applied to the member. 

So besides the `memberToAccessListMember` function, Access List `GET` and `PUT` apis along with Web UI needs to be updated to solve that issue.